openscenario/map_fragment/include/map_fragment/road_segment.hpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MAP_FRAGMENT__ROAD_SEGMENT__HPP_
+#define MAP_FRAGMENT__ROAD_SEGMENT__HPP_
+
+#include <lanelet2_core/geometry/Lanelet.h>
+
+#include <map_fragment/map_fragment.hpp>
+#include <map_fragment/parametric_curve.hpp>
+#include <rcpputils/asserts.hpp>
+
+namespace map_fragment
+{
+
+/**
+ * Description of a road cross section, defining its number of lanes and lane width
+ */
+struct RoadCrossSectionDescription
+{
+  const int number_of_lanes;
+
+  const double lane_width;
+
+  explicit RoadCrossSectionDescription(const int number_of_lanes, const double lane_width)
+  : number_of_lanes(number_of_lanes), lane_width(lane_width)
+  {
+    rcpputils::require_true(
+      number_of_lanes > 0,
+      "Expected number_of_lanes to be positive. Actual value: " + std::to_string(number_of_lanes));
+
+    rcpputils::require_true(
+      lane_width > 0.0,
+      "Expected lane_width to be positive. Actual value: " + std::to_string(lane_width));
+  }
+};  // struct RoadCrossSectionDescription
+
+/**
+ * Cross section of the road
+ */
+class RoadCrossSection
+{
+  lanelet::Points3d points_;
+
+public:
+  explicit RoadCrossSection(
+    const RoadCrossSectionDescription & description, const Point & origin,
+    const Vector & tangent_vector)
+  {
+    const auto n = description.number_of_lanes;
+    const auto w = description.lane_width;
```

Comment:
> This comes from my prejudice against long statements with long variable names. I will learn to fight it...

**I don't think this is your prejudice as it is simply a matter of preference. (Or perhaps more people prefer your way.)**
It is hardly a concern for the person who wrote the code, but for a new reader (i.e. me), it can be a pain to have to remember what the `n` and `w` stand for, even if only for a short period of time. It is like pushing new information into a small FIFO queue.

Reply:
> it would also make sense to rename i to something like lane_index. I didn't want to touch it at this point to maintain consistency with the rest of my code, where i is used for iterating through indices (of different things). I might later come back and change all occurrences at once.

There is no problem with this. Perhaps even I would do so.